### DISCUSSION TITLE
YouTube-DL links timing out (solution included)

### DISCUSSION THREAD

**Discussion by Kerosel** [savonet/liquidsoap#2152] [General]
I've been trying to drop some youtube-dl: links into my playback lately and it hasn't been working. The logs say that the process timed out. I did some research and found a solution.

Turns out back in June, YouTube/Google added a challenge/response in the ballet that is getting a downloadable link and if you do not respond correctly to the challenge, the connection is throttled down to like 50kB/s. This was causing a 6-minute song to take 2 minutes to download, which exceeds the default 30-second (or is it 20-second?) timeout period for external processes.

I found out there's a fork of youtube-dl called yt-dlp that has been updated to respond correctly to this challenge and bypass the throttling. That same 6-minute song came down in 2 seconds using yt-dlp (maybe 6 seconds total with the overhead to download the page and establish the link).

yt-dlp is available in the Debian repositories so I'm sure it's available for other Linuxes as well (I'm afraid I can't say if it's available for Windows, but I don't see why it shouldn't be). It's a drop-in replacement for youtube-dl so you just need to set a new youtube-dl executable at the start of your script:
```
settings.protocol.youtube_dl.path.set("yt-dlp")
```
Tested it this afternoon and it works! Links using the youtube-dl protocol in LiquidSoap no longer time out and queue up and play like they used to. If at some point youtube-dl is updated to step around the throttling, we can easily go back to it by removing that line.

**Comment by toots** [comment]
Great info, thank you very much!!
